sycl/source/detail/host_pipe_map_entry.hpp

```diff
@@ -24,7 +25,8 @@ struct HostPipeMapEntry {
   std::uint32_t MHostPipeTSize;
   // The device image that pipe is associated with
   const RTDeviceBinaryImage *mDeviceImage;
-
+  // Compiled PiProgram
+  RT::PiProgram mPiProgram;
```

Comment:
I have my reservations about this approach. What happens if the implementation chooses to evict the program from its cache? If that happens, you would potentially have a dead PI program here and you would have no way to check if it is dead.